collect_app/src/main/java/org/odk/collect/android/configure/qr/QRCodeScannerFragment.kt

```diff
@@ -30,19 +32,30 @@ class QRCodeScannerFragment : BarCodeScannerFragment() {
     @Inject
     lateinit var currentProjectProvider: CurrentProjectProvider
 
+    @Inject
+    lateinit var storagePathProvider: StoragePathProvider
+
     override fun onAttach(context: Context) {
         super.onAttach(context)
         DaggerUtils.getComponent(context).inject(this)
     }
 
     @Throws(IOException::class, DataFormatException::class)
     override fun handleScanningResult(result: BarcodeResult) {
+        val oldProjectName = currentProjectProvider.getCurrentProject().name
+
         val importSuccess = settingsImporter.fromJSON(
             CompressionUtils.decompress(result.text),
             currentProjectProvider.getCurrentProject()
         )
         val settingsHash = getMd5Hash(ByteArrayInputStream(result.text.toByteArray()))
         if (importSuccess) {
+            val newProjectName = currentProjectProvider.getCurrentProject().name
+            if (newProjectName != oldProjectName) {
```

Comment:
Is this going to compare string references? Should be .equals, no?

Reply:
It's kotlin and in kotlin reference equality is === not ==

Reply:
> It's kotlin

Sigh. It's tricky switching back and forth!